core/modules/entity/entity.controller.inc

```diff
@@ -181,16 +227,71 @@ class DefaultEntityController implements EntityControllerInterface {
     // and we need to know if it's empty for this reason to avoid querying the
     // database when all requested entities are loaded from cache.
     $passed_ids = !empty($ids) ? array_flip($ids) : FALSE;
+
+    // Use an entity field query to transform the list of conditions into
+    // the set of entity IDs which the conditions admit, then re-enter this
+    // method with that set as the $ids constraint.
+    if ($conditions && !$revision_id) {
```

Comment:
Noting that this PR will result in two DB queries now, if `$conditions` is set: this one to sort out the conditions and the next one, starting on line 295, to load the entities. But we have also made it clear that using `$conditions` to load entities is deprecated, so this would not be a problem for any core usage of `Entity::load()`, just any contrib that isnt paying attention.